File: samples/Test/Classes/Control/ImagePickerControllerTest.cpp

#include "ImagePickerControllerTest.h"

ImagePickerControllerTest::ImagePickerControllerTest()
{
    this->setTitle("ImagePickerControllerTest");
}

ImagePickerControllerTest::~ImagePickerControllerTest()
{
    
}

void ImagePickerControllerTest::viewDidLoad()
{
    
    CAButton* takePicture = CAButton::create(CAButton::Type::SquareRect);
    takePicture->setLayout(DLayout(DHorizontalLayout_W_C(240, 0.5), DVerticalLayout_H_C(54, 0.07)));
    takePicture->setTitleForState(CAControl::State::Normal,"拍照");
    takePicture->addTarget([=]()
    {
       // 打开相机
       CAImagePickerController* imgPicker = CAImagePickerController::create(CrossApp::CAImagePickerController::SourceType::CameraDeviceRear);
       imgPicker->open([=](CAImage* img)
       {
           CAImageView* image2 = CAImageView::createWithImage(img);
           image2->setLayout(DLayout(DHorizontalLayout_W_C(448, 0.5),DVerticalLayout_H_C(700, 0.6)));
           this->getView()->addSubview(image2);
       });
    }, CAButton::Event::TouchUpInSide);
    this->getView()->addSubview(takePicture);
    
    CAButton* photoAlbum = CAButton::create(CAButton::Type::SquareRect);
    photoAlbum->setLayout(DLayout(DHorizontalLayout_W_C(240, 0.5), DVerticalLayout_H_C(54, 0.15)));
    photoAlbum->setTitleForState(CAControl::State::Normal,"相册");
    photoAlbum->addTarget([=]()
    {
      // 打开相册
      CAImagePickerController* imgPicker = CAImagePickerController::create(CrossApp::CAImagePickerController::SourceType::PhotoLibrary);
      imgPicker->open([=](CAImage* img)
      {
          CAImageView* image2 = CAImageView::createWithImage(img);
          image2->setLayout(DLayout(DHorizontalLayout_W_C(448, 0.5), DVerticalLayout_H_C(700, 0.6)));
          this->getView()->addSubview(image2);
      });
    }, CAButton::Event::TouchUpInSide);
    this->getView()->addSubview(photoAlbum);
    
}

void ImagePickerControllerTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

File: samples/Test_JS/Resources/js/Control/ImagePickerControllerTest.js
/**
 * Created by zhanglei on 16/8/4.
 */
var ImagePickerControllerTest = ca.CAViewController.extend({
    ctor: function () {
        this._super();
    },
    viewDidLoad: function() {

         var _this_ = this;
         var takePicture = ca.CAButton.create(ca.CAButton.Type.SquareRect);
         takePicture.setLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(240, 0.5), ca.DVerticalLayout_H_C(54, 0.07)));
         //设置Button标题的显示文本和字体大小
         takePicture.setTitleForState(ca.CAControl.State.Normal, "拍照");
         //添加回调事件
         takePicture.addTarget(function(){
            var imgPicker = ca.CAImagePickerController.create(ca.CAImagePickerController.SourceType.CameraDeviceRear);
            imgPicker.open(function(img){
                           var img2 = ca.CAImageView.createWithImage(img);
                           img2.setLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(448, 0.5), ca.DVerticalLayout_H_C(700, 0.6)));
                            _this_.getView().addSubview(img2);
                           });
            }, ca.CAButton.Event.TouchUpInSide );
         this.getView().addSubview(takePicture);

        var photoAlbum = ca.CAButton.create(ca.CAButton.Type.SquareRect);
        photoAlbum.setLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(240, 0.5), ca.DVerticalLayout_H_C(54, 0.15)));
        //设置Button标题的显示文本和字体大小
        photoAlbum.setTitleForState(ca.CAControl.State.Normal, "相册");
        //添加回调事件
        photoAlbum.addTarget(function(){

            var imgPicker = ca.CAImagePickerController.create(ca.CAImagePickerController.SourceType.PhotoLibrary);
            imgPicker.open(function(img){
                var img2 = ca.CAImageView.createWithImage(img);
                img2.setLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(448, 0.5), ca.DVerticalLayout_H_C(700, 0.6)));
                _this_.getView().addSubview(img2);
            });
        }, ca.CAButton.Event.TouchUpInSide );
        this.getView().addSubview(photoAlbum);
    },
});
